packages/zod-nestjs/package.json

```diff
@@ -22,7 +22,8 @@
     "nestjs"
   ],
   "dependencies": {
-    "@anatine/zod-openapi": "^1.10.0"
+    "@anatine/zod-openapi": "^1.10.0",
+    "type-fest": "^3.3.0"
```

Comment:
Let's move type-fest completely to devDeps

```
  "dependencies": {
    "@anatine/zod-openapi": "^1.10.0"
  },
  "peerDependencies": {
    "zod": "^3.20.0",
    "openapi3-ts": "^2.0.0 || ^3.0.0",
    "@nestjs/core": "^7.0.0 || ^8.0.0 || ^9.0.0",
    "@nestjs/swagger": "^6.0.0"
  },
  "devDependencies": {
    "type-fest": "^3.3.0"
  }
```

Bundlers should already ignore it. But I think this will end up with less of a download footprint? If Nest is already using it, it will be hanging out in the node modules folder anyway. I could be wrong here though. 

It would be a breaking change, but I am also thinking it might be a good idea to make the `@anatine/zod-openapi` a peer dep instead of a hard dep. It adds an NPM install, but would provide better support and version management.

Reply:
I don't believe nest uses `type-fest` (looking at the dependencies, I don't see it). Since the type would be exposed to the user, it may need to be a dependency, not dev.
We can look into a way to do this _without_ `type-fest`. Maybe we can use some simplified version of `Merge` or something.

I think making `@anatine/zod-openapi` a peer dep is the right call, since the end user can use it to extend their types so they'd need to import it anyway.
On a side note, what is `openapi3-ts` used for? I didn't install it and the package seemed to work fine without it.

Reply:
Alternatively to the peer dep, we could just support the `extendApi` in the `createZodDto`. That should make it so the consumer never needs to use `@anatine/zod-openapi` if they are using nest, right?

```ts
export const createZodDto = <T extends OpenApiZodAny>(
  initialZodSchema: T,
  extendSchemaObject?: SchemaObject
): ZodDtoStatic<T> => {
  const zodSchema = extendApi(initialZodSchema, extendSchemaObject);
```